contrib/platform/src/com/sun/jna/platform/win32/Pdh.java

```diff
@@ -277,4 +277,122 @@
+     *            terminated by two NULL characters. Set to NULL if the
+     *            pcchCounterListLengthparameter is zero.
+     * @param pcchCounterListLength
+     *            Size of the mszCounterList buffer, in TCHARs. If zero on input
+     *            and the object exists, the function returns PDH_MORE_DATA and
+     *            sets this parameter to the required buffer size. If the buffer
+     *            is larger than the required size, the function sets this
+     *            parameter to the actual size of the buffer that was used. If
+     *            the specified size on input is greater than zero but less than
+     *            the required size, you should not rely on the returned size to
+     *            reallocate the buffer.
+     * @param mszInstanceList
+     *            Caller-allocated buffer that receives a list of
+     *            null-terminated instance names provided by the specified
+     *            object. The list contains unique instance names. The list is
+     *            terminated by two NULL characters. Set to NULL if
+     *            pcchInstanceListLength is zero.
+     * @param pcchInstanceListLength
+     *            Size of the mszInstanceList buffer, in TCHARs. If zero on
+     *            input and the object exists, the function returns
+     *            PDH_MORE_DATA and sets this parameter to the required buffer
+     *            size. If the buffer is larger than the required size, the
+     *            function sets this parameter to the actual size of the buffer
+     *            that was used. If the specified size on input is greater than
+     *            zero but less than the required size, you should not rely on
+     *            the returned size to reallocate the buffer. If the specified
+     *            object does not support variable instances, then the returned
+     *            value will be zero. If the specified object does support
+     *            variable instances, but does not currently have any instances,
+     *            then the value returned is 2, which is the size of an empty
+     *            MULTI_SZ list string.
+     * @param dwDetailLevel
+     *            Detail level of the performance items to return. All items
+     *            that are of the specified detail level or less will be
+     *            returned.
+     * @param dwFlags
+     *            This parameter must be zero.
+     * @return If the function succeeds, it returns ERROR_SUCCESS. If the
+     *         function fails, the return value is a system error code or a PDH
+     *         error code.
+     */
+    int PdhEnumObjectItems(String szDataSource, String szMachineName, String szObjectName, char[] mszCounterList,
+            DWORDByReference pcchCounterListLength, char[] mszInstanceList, DWORDByReference pcchInstanceListLength,
+            int dwDetailLevel, int dwFlags);
```

Comment:
The `char[]` parameters are problematic, as the type mapper will not change their mapping. A java char will be mapped to a wchar on the C side, but if `w32.ascii` is in effect, not `LPWSTR` is required (this is a `wchar[]`), but a `LPSTR' (a `char[]`).

So what was done in other places was this: map the function with a `byte[]` as parameter and add a `<Lib>Util` class, that hold a helper function, that unwraps the value correctly.

Reply:
Is there another example in the JNA project where this was done such that the Util class already exists?

Reply:
I think this is a good sample:

_Here is the MSDN definition_
https://msdn.microsoft.com/en-us/library/windows/desktop/aa376556(v=vs.85).aspx
_Binding_
https://github.com/java-native-access/jna/blob/master/contrib/platform/src/com/sun/jna/platform/win32/Crypt32.java#L359
_UtilHelper_
https://github.com/java-native-access/jna/blob/master/contrib/platform/src/com/sun/jna/platform/win32/Crypt32Util.java#L195

`psz` is an `LPTSTR` so it depends on which function variant is called (`CertNameToStrW` => unicode, `CertNameToStrA` => ascii).

Reply:
I think there are also some instances which explicitly declare both the `-A` and `-W` of a function.  The library invocation handler for `W32API` automatically tries appending the suffix if the explicit name does not exist, but if the explicit name _is_ used, that's what you'll get.  Then you can wrap those and choose in a util wrapper.